ruby.md

```diff
@@ -5,41 +5,42 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. In what ways are JavaScript and Ruby similar? In what ways are they different?
 
-  Your answer:
+  Your answer: Both are loosely typed, object oriented languages. Ruby has less falsy values since there is no type coercion. JS is mainly frontend as it can be used with React and Ruby is more backend because it can be used with Rails.
 
-  Researched answer:
+  Researched answer: Both are object oriented, dynamic, scripting languages. Ruby has simpler variable declaration while JS needs var, const, or let. JS is frontend while Ruby is backend. JS is faster than Ruby due to the fact that it has a more optimized enginge. JS is more scaleable than Ruby
 
 
 
 2. What is a hash?
 
-  Your answer:
+  Your answer: A hash is a dictionary or an object. It has key value pairs and can store them
 
-  Researched answer:
+  Researched answer: A Hash is a dictionary-like collection of unique keys and their values. Also called associative arrays, they are similar to Arrays, but where an Array uses integers as its index, a Hash allows you to use any object type.
 
 
 
 3. What is the testing framework used in Ruby? Describe the process of setting up the testing environment.
 
-  Your answer:
-
-  Researched answer:
+  Your answer: RSPEC. You need to install the RPSEC gem, require RSPEC in your file then run it in your terminal through RSPEC file_name
 
+  Researched answer: RSpec is a 'Domain Specific Language' (DSL) testing tool written in Ruby to test Ruby code.[4] It is a behavior-driven development (BDD) framework which is extensively used in production applications.
 
 
 4. Name three possible relationships between objects?
 
```

Comment:
You are correct, but the terminology is more specific when creating the relationships in your code. is-a, has-many, etc